### PR TITLE
Expression

### PR DESCRIPTION
**Pull request by zwospiel**
Only tests.

### CODE REVIEW

**Comment by zwospiel** [comment]
One test is failing, because the test for InvalidInput Error occurs before the UnknownOperator Error, not sure what to do with that. All other tests are succeeding.

File: tests/Expression.test.js [hunk 1]
```diff
@@ -35,7 +35,78 @@ describe("Expression.constructor", () => {
...
+            expression = new Expression("3*+4")
+            expect(() => expression.solve()).toThrow(MalformedExpression)
+            expression = new Expression("7**+*+5")
+            expect(() => expression.solve()).toThrow(MalformedExpression)
+        })
+        test("for operator at the beginning", () => {
+            let expression = new Expression("+3*4")
+            expect(() => expression.solve()).toThrow(MalformedExpression)
+        })
+        test("for operator at the beginning preceded by whitespaces", () => {
```
Comment (c-auri) [inline tests/Expression.test.js:75]: This test cast can be moved down into the "ignores whitespace" group.

File: tests/Expression.test.js [hunk 2]
```diff
@@ -35,7 +35,78 @@ describe("Expression.constructor", () => {
...
+            let expression = new Expression("    +3*4")
+            expect(() => expression.solve()).toThrow(MalformedExpression)
+        })
+        test("for operator at the end", () => {
+            let expression = new Expression("3*4+")
+            expect(() => expression.solve()).toThrow(MalformedExpression)
+        })
+        test("for operator at the end followed by whitespaces", () => {
```
Comment (c-auri) [inline tests/Expression.test.js:83]: This test cast can be moved down into the "ignores whitespace" group.

File: src/js/Expression.js [hunk 3]
```diff
@@ -68,13 +70,61 @@ export class Expression {
...
+                this.#validateTokenOrder(number)
                 yield number
-            } else if (this.#input[i] === " ") {
-                continue
-            } else {
+            } else if (isOperator(this.#input[i]) || isBracket(this.#input[i])) {
+                this.#validateTokenOrder(this.#input[i])
                 yield this.#input[i]
+            } else {
+                throw new InvalidInput(this.#input)
```
Comment (c-auri) [inline src/js/Expression.js:79]: Did we decide to change something about the `InvalidInput` error in `Operators.apply`? I don't remember exactly.

**Review by c-auri — CHANGES_REQUESTED** [review]
Let's talk about the minor issues in the next meeting.

**Review by c-auri — APPROVED** [review]

**Pull request merged**